legendary_eureka.py

```diff
@@ -34,6 +39,99 @@ class Glassdoor(Resource):
+            'Cache-Control': 'no-cache',
+            'Connection': 'keep-alive'
+        }
+        data = {"term": 'Israel',
+                "maxLocationsToReturn": 10}
+
+        location_url = "https://www.glassdoor.co.in/findPopularLocationAjax.htm?"
+
+        # Getting location id for search location
+        print("Fetching location details")
+        location_response = requests.post(location_url, headers=location_headers, data=data).json()
+
+        if job_location in [dictionary['label'] for dictionary in location_response]:
+            location_response = [dictionary for dictionary in location_response
+                                 if job_location in dictionary['label']]
+        else:
+            print('Job location not found, printing jobs from all locations in Israel..')
+
+        for i, location in enumerate(location_response):
+            job_listings = {}
+            place_id = location_response[i]['locationId']
+            place_label = location_response[i]['label']
+
+            if place_id:
+                print("Fetching jobs for location: ", place_label)
+
+                #  delay time for avoiding block by requests.post
+                #  TODO: change delay time with better (and faster) anti-block strategy.
+                if i > 0:
+                    delays = [3, 2, 1]
+                    delay = random.choice(delays)
+                    time.sleep(delay)
+
+                job_listing_url = 'https://www.glassdoor.com/Job/jobs.htm'
+
+                # Form data to get job results
+                data = {
+                    'clickSource': 'searchBtn',
+                    'sc.keyword': job_title,
+                    'locT': 'C',
+                    'locId': place_id,
+                    'jobType': job_type
+                }
+
+                response = requests.post(job_listing_url, headers=headers, data=data)
+                soup = BeautifulSoup(response.content, "html.parser")
+
+                for a in soup.find_all('a', href=True):
+
+                    if a['href'].startswith('/partner/'):
+                        link = a['href']
+                        full_link = 'https://www.glassdoor.com'+link  # TODO: shrink full links length with tiny url?
+                        job_id = link.split('jobListingId=')[1]
+
+                        if (a.string is not None) and (a.string != 'JobInfo'):
+
+                            if place_label in job_listings:
+                                if job_id not in job_listings[place_label]:
+                                    job_listings[place_label].update({job_id: [a.string, full_link]})
+                                else:
+                                    if a.string not in job_listings[place_label][job_id]:
+                                        job_listings[place_label][job_id].append(a.string)
+                            else:
+                                job_listings[place_label] = {job_id: [a.string, full_link]}
+
```

Comment:
Maybe use `switch case statement` instead of all of those nested `if`s
[How to implement a switch-case statement in Python?](https://jaxenter.com/implement-switch-case-statement-python-138315.html)

Reply:
```python
    if a['href'].startswith('/partner/') and \
        (a.string is not None) and \
        (a.string != 'JobInfo'):

        link = a['href']
        full_link = 'https://www.glassdoor.com'+link 
        job_id = link.split('jobListingId=')[1]
        if (place_label in job_listings) and (job_id not in job_listings[place_label]):
             job_listings[place_label].update({job_id: [a.string, full_link]})
        elif (place_label in job_listings) and (a.string not in job_listings[place_label][job_id]):
             job_listings[place_label][job_id].append(a.string)
        else:
            job_listings[place_label] = {job_id: [a.string, full_link]}
```
Or using this snippet